pkg/operator/ceph/nfs/spec.go

```diff
@@ -246,9 +247,29 @@ func (r *ReconcileCephNFS) daemonContainer(nfs *cephv1.CephNFS, cfg daemonConfig
 		Env:             controller.DaemonEnvVars(r.cephClusterSpec),
 		Resources:       nfs.Spec.Server.Resources,
 		SecurityContext: controller.PodSecurityContext(),
+		LivenessProbe:   r.genGaneshaLivenessProbe(nfs),
 	}
 }
 
+func (r *ReconcileCephNFS) genGaneshaLivenessProbe(nfs *cephv1.CephNFS) *v1.Probe {
+	if nfs.Spec.Server.LivenessProbe == nil || nfs.Spec.Server.LivenessProbe.Disabled {
+		// without liveness-probe
+		return nil
+	}
+	if nfs.Spec.Server.LivenessProbe.Probe != nil {
+		// liveness-probe using user-defined probe
+		return nfs.Spec.Server.LivenessProbe.Probe
```

Comment:
For other probes, we allow users to "override" portions of the default probe. This is achieved using this method:
https://github.com/rook/rook/blob/929bc9720f45b0752fec912a69f5a7d58d9066cd/pkg/operator/ceph/config/probes.go#L64
 
Using the method is a little awkward tbh (current probe is the 'default'), but I think we should keep the same behavior here as elsewhere so it's consistent within Rook.

Reply:
ACK